pinot-core/src/test/java/com/linkedin/pinot/core/realtime/kafka/SimpleConsumerWrapperTest.java

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.core.realtime.kafka;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.pinot.core.realtime.impl.kafka.KafkaSimpleConsumerFactory;
+import com.linkedin.pinot.core.realtime.impl.kafka.SimpleConsumerWrapper;
+import java.util.Collections;
+import java.util.HashMap;
+import kafka.api.FetchRequest;
+import kafka.api.PartitionFetchInfo;
+import kafka.api.PartitionMetadata;
+import kafka.api.TopicMetadata;
+import kafka.cluster.Broker;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.FetchResponse;
+import kafka.javaapi.TopicMetadataRequest;
+import kafka.javaapi.TopicMetadataResponse;
+import kafka.javaapi.consumer.SimpleConsumer;
+import kafka.javaapi.message.ByteBufferMessageSet;
+import kafka.message.Message;
+import org.apache.kafka.common.protocol.Errors;
+import org.testng.annotations.Test;
+import scala.Some;
+import scala.Tuple2;
+import scala.collection.Seq;
+import scala.collection.immutable.List;
+
+import static org.testng.Assert.assertEquals;
+
+
+/**
+ * Tests for the SimpleConsumerWrapper.
+ */
+public class SimpleConsumerWrapperTest {
+  public class MockKafkaSimpleConsumerFactory implements KafkaSimpleConsumerFactory {
+    public MockKafkaSimpleConsumerFactory(String[] hosts, int[] ports, long[] partitionStartOffsets,
+        long[] partitionEndOffsets, int[] partitionLeaderIndices, String topicName) {
+      Preconditions.checkArgument(hosts.length == ports.length);
```

Comment:
you could even use Assert instead of Preconditions.check() ? anyone using these mock objects should conform to what it implements ...